program-runtime/src/compute_budget.rs

```diff
@@ -276,6 +284,15 @@ impl ComputeBudget {
             .map(|fee_type| PrioritizationFeeDetails::new(fee_type, self.compute_unit_limit))
             .unwrap_or_default())
     }
+
+    // https://github.com/solana-labs/solana/issues/31453 to round compute-unit-price, in micro-lamports,
+    // down to its nearest 1_000 micro-lamport.
+    fn round_prioritization_fee(micro_lamports: u64) -> u64 {
+        let round_down_factor = MICRO_LAMPORTS_PER_LAMPORT.saturating_div(1_000);
```

Comment:
This indirection is **really** misleading, and seems to work only because of happenstance.
The `round_down_factor` should just be 1_000 as a constant.

If we were rounding to the nearest 10_000 micro-lamports, the factor should be 10_000, not 100, which this calculation would give.

Reply:
ah, i have over-played "1_000 micro-lamport" here 😜 Taking your suggestion.

Reply:
5cf0e180ac4d05ab8a874d848da61c355793dcf8 , also saved from exporting `MICRO_LAMPORTS_PER_LAMPORT`